src/cache.rs

```diff
@@ -341,6 +341,31 @@ impl Cache {
                 }
             })
             .collect::<Vec<String>>();
+
+        if let Some(custom_pages_dir) = custom_pages_dir {
+            let is_page = |entry: &DirEntry| -> bool {
+                let path = entry.path();
+                let extension = &path.extension().and_then(OsStr::to_str).unwrap_or("");
+                entry.file_type().is_file() && extension == &"page"
+            };
+            let to_stem = |entry: DirEntry| -> Option<String> {
+                entry
+                    .path()
+                    .file_stem()
+                    .and_then(|stem| stem.to_str().map(Into::into))
+            };
+            let mut custom_pages = WalkDir::new(custom_pages_dir)
+                .min_depth(1)
+                .max_depth(1)
+                .into_iter()
+                .filter_entry(is_page)
+                .filter_map(Result::ok)
+                .filter_map(to_stem)
+                .collect::<Vec<String>>();
+
+            pages.append(&mut custom_pages);
```

Comment:
This is a little anti-pattern: We collect into a vector, only to consume the elements right afterwards. We can let `custom_pages` just be the iterator we construct (by _not_ `collect`ing) and extend `pages` from that using `pages.extend(custom_pages);`. This way, we don't have the overhead of allocating the vector :)